### PR TITLE
Credit page

### LINKED ISSUE #498: Credits Page
- create a separate page for contributors containing 3 different features.
    * Tabs component to navigate between the teams
    * card info for each contributor containing his name, title, contacts and his contributions
    * stats about his progress in pickify
- most of these features components has already been made before with little adjustments, so please try to utilize them

### PR DESCRIPTION
**Pull request by atwfk**
## closes #498

- prepared **TabGroup** component to reuse it in **Credits Page**
- created **Credit** & **Credits** components & **Credits Page** with storybook presentation
- added dummy data for each mentee

## ToDo
- add tests for **Credit** & **Credits** components
- add real feedback from our mentors to each mentee

### CODE REVIEW

**Review by ahmedelsayad11 — CHANGES_REQUESTED** [review]
the post creation [tabgroup] is broken and does not work as expected

File: modules/CreditsPage/components/Credit/Credit.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+              <ImgWithInfo.Info classes="h-full">
+                <>
+                  <ImgWithInfo.Info.Title titleSize="medium">
+                    {creditData.developerName}
+                  </ImgWithInfo.Info.Title>
+                  <ImgWithInfo.Info.SubTitle subTitleSize="small">
+                    {creditData.developerTitle}
+                  </ImgWithInfo.Info.SubTitle>
+                  <ImgWithInfo.Info.MoreInfo>
+                    <ul className="list-none flex">
```
Comment (omarawad0) [inline modules/CreditsPage/components/Credit/Credit.tsx:34]: clicking on links should navigate the user to a new tab and keep Pickify's tab active

File: modules/CreditsPage/components/Credit/Credit.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          <div className="flex flex-col items-center">
+            <DeticatedRank className="mb-2" />
+            <span className="text-xs font-light text-dark">
+              {creditData.developerRank}
+            </span>
+          </div>
+        </>
+      </Box.Header>
+      <Box.Footer withDevider>
+        <div className="bg-grey-bg p-3">
```
Comment (omarawad0) [inline modules/CreditsPage/components/Credit/Credit.tsx:66]: please add a border-radius of 8px to the feedback's grey background.

**Review by omarawad0 — CHANGES_REQUESTED** [review]
Good work Tawfiek.
please consider these required changes.

File: modules/shared/components/molecules/Footer/footerData.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export const footerData = [
     path: '/',
     content: 'About',
   },
+  { name: 'pickify-team', path: '/credits', content: 'Pickify Team' },
```
Comment (omarawad0) [inline modules/shared/components/molecules/Footer/footerData.ts:7]: rename it just Team.

**Comment by omarawad0** [comment]
Can we make this page accessible from the login page without having to be logged in?

File: modules/CreditsPage/pages/CreditsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  const changeCurrentSelectedTab = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
+    setCurrentSelectedTab(e.currentTarget.value);
+  };
+
+  let creditsData = [];
+
+  if (currentSelectedTab === 'Front-end') {
```
[the remark this answers is not in the document]
Comment (atwfk) [reply modules/CreditsPage/pages/CreditsPage.tsx:23]: useEffect hook gets executed asynchronously after rendering the DOM tree, so we can't initialize the teams data inside

[the remark this answers is not in the document]
Comment (omarawad0) [reply modules/CreditsPage/pages/CreditsPage.tsx:23]: I don't see any problem with this, just remove the let variable and create a state instead. and make the useEffect synchronized with the currentSelectedTab state

**Review by ahmedelsayad11 — APPROVED** [review]

**Pull request merged**